src/harness/renderers/gl/gl_renderer.c

```diff
@@ -389,6 +389,10 @@ void GLRenderer_EndScene() {
 
     generated_current_framebuffer_for_this_frame = 0;
     CHECK_GL_ERROR("GLRenderer_EndScene");
+
+    /* Fixes blending in map mode, which renders the scene thrice per frame. */
+    glBindFramebuffer(GL_FRAMEBUFFER, framebuffer_id);
```

Comment:
this has the effect of breaking shadow rendering. Instead, I'm happy to wrap the for loop in `RenderAFrame` in a `#if DETHRACE_FIX_BUGS`, which should also resolve the issue?

I've looked and looked and can't see any reason for it. My guess is it was some leftover from a debugging session that made it into the retail build as an oversight

Reply:
Yes, the issue is gone if we only loop once. I also spent some time trying to understand the logic behind the loop and got nothing. So I'm fine with the workaround you're proposing.
Have we tried to hack CARMA95.EXE, so it only loops once, to see if anything breaks there?

Reply:
Interestingly, in `VOODOO2C.exe`, its slightly different:

```
v27 = !gMap_mode || gSome_var_only_set_in_voodoo_rendering_mode ? 1 : 3
```

So if it is in voodoo mode, the triple rendering doesn't happen. Its weird though - why go to the trouble of updating the check instead of removing the loop entirely? 🤔  

/ @madebr

Reply:
yes, I was curious again so just tried it (I updated the linked issue #143). Couldn't see anything broken :/